SofaKernel/modules/SofaSimulationGraph/DAGNode.cpp

```diff
@@ -330,12 +330,13 @@ void DAGNode::getObjects(const sofa::core::objectmodel::ClassInfo& class_info, G
         case Local:
             this->getLocalObjects( class_info, container, tags );
             break;
-
         case SearchUp:
             this->getLocalObjects( class_info, container, tags ); // add locals then SearchParents
-            // no break here, we want to execute the SearchParents code.
+            break;
         case SearchParents:
         {
+            this->getLocalObjects( class_info, container, tags ); // add locals then SearchParents
```

Comment:
I came upon this while randomly looking at open merge requests: this change does look wrong to me.
The original code was using the fact that SearchUp is supposed to search for local objects and then within parents, while SearchParents should only search within parent. Without the break, both cases could share the same code for the shared part.
The changed code is no longer looking in parents for SearchUp while it should, and is searching for local objects for SearchParents while it should not...
I suspect this was done because of the gcc warning about implicit-fallthrough. There are several ways to silence that warning, depending on the c++ standard that should be supported (see [here](http://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wimplicit-fallthrough_003d)). The easiest is to add a comment `/* FALLTHRU */` above the `case`.

Reply:
You are defintely right, thanks for reporting.